alicloud/resource_alicloud_polardb_cluster.go

```diff
@@ -0,0 +1,454 @@
+			d.SetId("")
+			return nil
+		}
+		return WrapError(err)
+	}
+
+	ips, err := polarDBService.GetSecurityIps(d.Id())
+	if err != nil {
+		return WrapError(err)
+	}
+
+	d.Set("security_ips", ips)
+
+	d.Set("vswitch_id", clusterAttribute.VSwitchId)
+	d.Set("pay_type", clusterAttribute.PayType)
+	d.Set("id", clusterAttribute.DBClusterId)
+	d.Set("description", clusterAttribute.DBClusterDescription)
+	d.Set("db_type", clusterAttribute.DBType)
+	d.Set("db_version", clusterAttribute.DBVersion)
+	d.Set("maintain_time", clusterAttribute.MaintainTime)
+	d.Set("zone_ids", clusterAttribute.ZoneIds)
+	d.Set("cluster_charge_type", getChargeType(clusterAttribute.PayType))
+	d.Set("cluster_network_type", clusterAttribute.DBClusterNetworkType)
+	d.Set("db_node_class", cluster.DBNodeClass)
+
+	if clusterAttribute.PayType == string(Prepaid) {
+		clusterAutoRenew, err := polarDBService.DescribePolarDBAutoRenewAttribute(d.Id())
+		if err != nil {
+			if NotFoundError(err) || IsExceptedErrors(err, []string{InvalidDBClusterIdNotFound, InvalidDBClusterNameNotFound}) {
+				d.SetId("")
+				return nil
+			}
+			return WrapError(err)
+		}
+		d.Set("auto_renew", clusterAutoRenew.AutoRenewEnabled)
+		renewPeriod := 1
+		if clusterAutoRenew != nil {
+			renewPeriod = clusterAutoRenew.Duration
+		}
+		if clusterAutoRenew != nil && clusterAutoRenew.PeriodUnit == string(Year) {
+			renewPeriod = renewPeriod * 12
+		}
+		d.Set("auto_renew_period", renewPeriod)
+		d.Set("period", computePeriodByMonth(clusterAttribute.CreationTime, clusterAttribute.ExpireTime))
+		d.Set("renewal_status", clusterAutoRenew.RenewalStatus)
+	}
+
+	if err = polarDBService.RefreshParameters(d, "parameters"); err != nil {
+		return WrapError(err)
+	}
+
+	if clusterAttribute.PayType == string(Prepaid) {
+		request := polardb.CreateDescribeAutoRenewAttributeRequest()
+		request.RegionId = client.RegionId
+		request.DBClusterIds = d.Id()
+
+		raw, err := client.WithPolarDBClient(func(polarDBClient *polardb.Client) (interface{}, error) {
+			return polarDBClient.DescribeAutoRenewAttribute(request)
+		})
+		if err != nil {
+			return WrapErrorf(err, DefaultErrorMsg, d.Id(), request.GetActionName(), AlibabaCloudSdkGoERROR)
+		}
+		addDebug(request.GetActionName(), raw, request.RpcRequest, request)
+		response, _ := raw.(*polardb.DescribeAutoRenewAttributeResponse)
+		if response != nil && len(response.Items.AutoRenewAttribute) > 0 {
+			renew := response.Items.AutoRenewAttribute[0]
+			d.Set("auto_renew", renew.AutoRenewEnabled)
+			d.Set("auto_renew_period", renew.Duration)
+		}
+	}
+
+	return nil
+}
+
+func resourceAlicloudPolarDBClusterDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	polarDBService := PolarDBService{client}
+
+	cluster, err := polarDBService.DescribePolarDBClusterAttribute(d.Id())
+	if err != nil {
+		if NotFoundError(err) || IsExceptedErrors(err, []string{InvalidDBClusterIdNotFound, InvalidDBClusterNameNotFound}) {
+			return nil
```

Comment:
I think the line 354-357 are redundant. It the cluster is not found, the Delete API will return a not found error and the next step will return nil